test/test_retrieve_utils.py

```diff
@@ -100,6 +100,67 @@ def test_query_vector_db(self):
         results = query_vector_db(["autogen"], client=client)
         assert isinstance(results, dict) and any("autogen" in res[0].lower() for res in results.get("documents", []))
 
+    def test_custom_vector_db(self):
+        import lancedb
```

Comment:
consider the case where lancedb is not installed.

Reply:
> consider the case where lancedb is not installed.

It's a test. Why would that happen?

Reply:
Addressed.